ERCS/erc-7690.md

```diff
@@ -0,0 +1,129 @@
+
+interface IERC7690 {
+
+    function getFunctionCallTimes(bytes4 selector) external view returns (uint256);
+
+    function getContractCallTimes() external view returns (uint256);
+}
+```
+
+`getFunctionCallTimes`: Allows cross-contract acquisition of the number of times a function is called in a transaction.
+
+`getContractCallTimes`: Allows cross-contract acquisition of the number of times a contract is called in a transaction.
+
+## Rationale
+
+The attack principle of flash loan is to inject a large amount of assets into a contract that a DApp relies on to change the data of the contract, so that the DApp can be lent or obtained more assets. 
+
+The main attack modes of flash loans are:
+
+1. Lending a large amount of assets at low interest rates, and then injecting funds into the data contract (group), which is the dependency of the attacked contract (group).
+2. Obtain a large amount of funds from the attacked contract (group). The contract (group) being attacked may be a data contract + other contracts that produce funds. For example, in an exchange, first inject borrowed assets into the exchange to lower/increase the price of an asset, then attack other contracts that depend on it, and finally obtain the previously borrowed assets from the exchange.
+3. After the attack is completed, repay the obtained assets to the flash loan.
+
+For example, in the example of a lending DApp that relies on the current price of a decentralized exchange, the attacker borrows assets through flash loans, injects them into the decentralized exchange to change the price, and then borrows another asset from the DApp based on the mortgage asset. Then the price on the decentralized exchange is restored, and the loaned asset portion will be lower than the mortgage asset portion, thereby realizing flash loan attack arbitrage.
+
+The reasons why flash loans can be attacked are:
+
+1. A function does not know the number of times any other function is called in a transaction
+2. A contract will not know the number of times any other contract has been called in a transaction.
+
+Union Lock can solve this problem.
+
+Union Lock allows counting the number of any function calls in a transaction. In the above example, when the injected function of the decentralized exchange obtained is not 0, the transaction can be rejected to avoid flash loan attacks.
+
+This is all based on the fact that Union Lock can count the number of contract calls and function calls with TSTORE, and if it is accessed across contracts in the same transaction, it will be automatically reset to zero in the next transaction.
+
+## Backwards Compatibility
+
+This EIP requires the implementation of TSTORE/TLOAD.
+
+Since this EIP does not change behavior of any existing opcodes, it is backwards compatible with all existing smart contracts.
+
+## Test Cases
+
+Reference @1999321 Union-Lock repository
+
+## Reference Implementation
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Same license comment.

Reply:
change it to "// CC0 1.0 Universal (CC0 1.0) Public Domain Dedication"